File: app/src/main/java/com/airbnb/net/ListItem.java
package com.airbnb.net;

/**
 * A list item that can either contain a {@link HeroItem} or a {@link ListItem}.
 */
public class ListItem {

    public HeroItem heroItem;
    public ListingItem listingItem;

    public ListItem(HeroItem heroItem) {
        this.heroItem = heroItem;
    }

    public ListItem(ListingItem listingItem) {
        this.listingItem = listingItem;
    }
}

File: app/src/main/java/com/airbnb/net/ListItem.kt
package com.airbnb.net

/**
 * A list item that can either contain a [HeroItem] or a [ListItem].
 */
public class ListItem {

    public var heroItem: HeroItem? = null
    public var listingItem: ListingItem? = null

    public constructor(heroItem: HeroItem) {
        this.heroItem = heroItem
    }

    public constructor(listingItem: ListingItem) {
        this.listingItem = listingItem
    }
}
